src/Uno.Toolkit.UI/Controls/NavigationBar/Renderer.cs

```diff
@@ -48,6 +48,13 @@ public Renderer(TElement element)
 				throw new ArgumentNullException(nameof(element));
 			}
 
+#if __IOS__
+			if (element is FrameworkElement fe)
+			{
+				fe.Unloaded += (s, e) => Dispose();
+			}
+#endif
```

Comment:
For this, we needed to ensure that any registered renderers do not attempt to Render() if they are on a Page that is not currently displayed. The iOS nativeframepresenter will re-wire up the renderers when a page is about to appear

But this'd be holding a ref to Element now instead of just the WeakReference. So I don't know if this is a good enough workaround

Reply:
You may want to remove the presenter after the dispose has been run to avoid the leak?